docs/docs/environment-variables.md

```diff
@@ -106,7 +106,7 @@ render() {
 }
 ```
 
-`API_KEY` will be available to your site (Server-side) as `process.env.API_KEY`. If you commit your `.env.*` file containing `API_KEY` to source control it would also be available on the client-side. However we **strongly** advise against that! You should prefix your variable with `GATSBY_` (as shown above) instead and Gatsby automatically makes it available in the browser context.
+In Node, `API_KEY` will be available to your site (Server-side) as `process.env.API_KEY`. You can use `dotenv` to commit your `.env.*` file containing `API_KEY` in order to make it available client-side, but we recommend against checking these files into source control. By prefixing your variable with `GATSBY_` (as shown above), Gatsby automatically embeds it as process.env.GATSBY_* in compiled JS making it available in the browser context.
```

Comment:
```suggestion
In Node, your site has access to `API_KEY` within the context of Gatsby build (Server-side) using the identifier `process.env.API_KEY`. In order to access this information client-side you can create a `.env.*` file containing `API_KEY`, however, we recommend against checking these files into source control as it is a security risk to expose your key. The safer alternative is to prefix your variable with `GATSBY_` (as shown above) in your Gatsby build configuration. This will prompt Gatsby to automatically embed the variable as process.env.GATSBY_* in compiled JS making it available in the browser context without exposing it elsewhere.
```